src/network/networkd-link.c

```diff
@@ -1161,6 +1161,16 @@ static int link_set_flags(Link *link) {
                 SET_FLAG(ifi_flags, IFF_ALLMULTI, link->network->allmulticast);
         }
 
+        if (link->network->promiscuous >= 0) {
+                ifi_change |= IFF_PROMISC;
+                SET_FLAG(ifi_flags, IFF_PROMISC, link->network->promiscuous);
+        }
+
+        if (link->network->trailers >= 0) {
+                ifi_change |= IFF_NOTRAILERS;
+                SET_FLAG(ifi_flags, IFF_NOTRAILERS, link->network->trailers);
```

Comment:
And negate the value: `SET_FLAG(ifi_flags, IFF_NOTRAILERS, !link->network->trailers);`